src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -758,6 +764,55 @@ export class ResultsViewPageStore extends AnalysisStore
         }
     }
 
+    /**
+     * Genes of 'single gene' queries
+     * (i.e. non-fusion queries, without an upstream or downstream gene)
+     */
+    @computed get singleHugoGeneSymbols() {
+        if (
+            this.urlWrapper.query.gene_list &&
+            this.urlWrapper.query.gene_list.length > 0
+        ) {
+            return nonFusionsInOQLQuery(this.urlWrapper.query.gene_list);
+        } else {
+            return [];
+        }
+    }
+
+    @computed get structVarQueries(): SingleGeneQuery[] {
+        if (
+            this.urlWrapper.query.gene_list &&
+            this.urlWrapper.query.gene_list.length > 0
+        ) {
+            return fusionsInOQLQuery(this.urlWrapper.query.gene_list);
+        } else {
+            return [];
+        }
+    }
+
+    /**
+     * All genes ($.gene and $.alterations.gene) of fusion queries
+     * (i.e. queries with an upstream or downstream gene or special value)
+     */
+    @computed get structVarHugoGeneSymbols(): string[] {
+        return _(this.structVarQueries)
+            .flatMap(query => [getFirstGene(query), getSecondGene(query)])
+            .compact()
```

Comment:
unless i'm mistaken, compact and uniq will have no impact here.  flatMap will return a n array of arrays (regardless if contents of child arrays are undefined).  The child arrays will always be A) defined and B) unique.

Reply:
No, as far as I can tell for the lodash docs flatMap returns an array. The chained .compart() will remove any falsy values.

Reply:
Flatmap will return an array of whatever the callback/iteratee returns.  Here you are returning an array, so it will be an array of arrays, none of which will be undefined or duplicated.
![image](https://user-images.githubusercontent.com/186521/217900220-8453d3c5-09a6-42a7-a656-5fbce3b43c9e.png)

Reply:
No, your example here is not correct. To match the code it should be:

```
_.flatMap([[1,2], [3,4]], m => [m[0], m[1]])
```

Moreover, I know I am right because 1) the code works and 2) the type checking agress with the specified `string[]` contract.